### ISSUE TITLE
Random test failure while renaming a folder

### ISSUE THREAD

**Issue by deivid-rodriguez** [rubygems/rubygems#3568]
I noticed the following random test failure: https://github.com/rubygems/rubygems/runs/627790381.

The error reads as follow:

```
Failures:

  1) bundle install --standalone with gemcutter's dependency API simple gems works on a different system
     Failure/Error: FileUtils.mv(bundled_app, "#{bundled_app}2")

     Errno::ENOTEMPTY:
       Directory not empty @ dir_s_rmdir - d:/a/rubygems/rubygems/bundler/tmp/2/bundled_app

       Commands:
       $  D:/rubyinstaller-2.5.8-1-x64/bin/ruby.exe -Id:/a/rubygems/rubygems/bundler/lib;d:/a/rubygems/rubygems/bundler/spec -rsupport/hax -rsupport/artifice/endpoint d:/a/rubygems/rubygems/bundler/exe/bundle install --path d:/a/rubygems/rubygems/bundler/tmp/2/bundled_app/bundle --standalone
       Fetching gem metadata from http://localgemserver.test/....
       Resolving dependencies...
       Fetching rake 13.0.1
       Installing rake 13.0.1
       Using bundler 2.2.0.dev
       Fetching activesupport 2.3.2
       Installing activesupport 2.3.2
       Fetching actionmailer 2.3.2
       Fetching actionpack 2.3.2
       Installing actionpack 2.3.2
       Installing actionmailer 2.3.2
       Fetching activerecord 2.3.2
       Fetching activeresource 2.3.2
       Installing activeresource 2.3.2
       Installing activerecord 2.3.2
       Fetching rails 2.3.2
       Installing rails 2.3.2
       Bundle complete! 1 Gemfile dependency, 8 gems now installed.
       Bundled gems are installed into `./bundle`
       [DEPRECATED] The `--path` flag is deprecated because it relies on being remembered across bundler invocations, which bundler will no longer do in future versions. Instead please use `bundle config set path 'd:/a/rubygems/rubygems/bundler/tmp/2/bundled_app/bundle'`, and stop using this flag
       # $? => 0
     Shared Example Group: "common functionality" called from ./spec/install/gems/standalone_spec.rb:243
     Shared Example Group: "bundle install --standalone" called from ./spec/install/gems/standalone_spec.rb:301
     # ./spec/install/gems/standalone_spec.rb:30:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:112:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:112:in `block (2 levels) in <top (required)>'
     # ./spec/spec_helper.rb:79:in `block (2 levels) in <top (required)>'
     # ./spec/support/rubygems_ext.rb:91:in `load'
     # ./spec/support/rubygems_ext.rb:91:in `gem_load_and_activate'
     # ./spec/support/rubygems_ext.rb:20:in `gem_load'
     # ------------------
     # --- Caused by: ---
     # Errno::EXDEV:
     #   Improper link @ rb_file_s_rename - (d:/a/rubygems/rubygems/bundler/tmp/2/bundled_app, d:/a/rubygems/rubygems/bundler/tmp/2/bundled_app2)
     #   ./spec/install/gems/standalone_spec.rb:30:in `block (3 levels) in <top (required)>'

Finished in 37 minutes 43 seconds (files took 0 seconds to load)
2647 examples, 1 failure, 179 pending
```

**Comment by deivid-rodriguez** [comment]
Another one, but on a different place: https://github.com/rubygems/rubygems/pull/3561/checks?check_run_id=629139681.

```
Failures:

  1) bundle binstubs <gem> when the gem exists in the lockfile the bundle binstub using another binstub loads all gems
     Failure/Error: FileUtils.mv root + File.basename(bundler_path), bundler_path

     Errno::ENOENT:
       No such file or directory @ rb_file_s_rename - (/home/runner/work/rubygems/rubygems/bundler/bundler-2.2.0.dev.gem, /home/runner/work/rubygems/rubygems/bundler/tmp/2/bundler-2.2.0.dev.gem)

       Commands:
       $ gem build /home/runner/work/rubygems/rubygems/bundler/bundler.gemspec
       Successfully built RubyGem
         Name: bundler
         Version: 2.2.0.dev
         File: bundler-2.2.0.dev.gem
       # $? => 0
     # ./spec/support/helpers.rb:310:in `block in with_built_bundler'
     # ./spec/support/path.rb:172:in `with_root_gemspec'
     # ./spec/support/helpers.rb:305:in `with_built_bundler'
     # ./spec/support/helpers.rb:287:in `block in install_gems'
     # ./spec/support/helpers.rb:285:in `each'
     # ./spec/support/helpers.rb:285:in `install_gems'
     # ./spec/support/helpers.rb:387:in `system_gems'
     # ./spec/commands/binstubs_spec.rb:102:in `block (4 levels) in <top (required)>'
     # ./spec/spec_helper.rb:112:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:112:in `block (2 levels) in <top (required)>'
     # ./spec/spec_helper.rb:79:in `block (2 levels) in <top (required)>'
     # ./spec/support/rubygems_ext.rb:91:in `load'
     # ./spec/support/rubygems_ext.rb:91:in `gem_load_and_activate'
     # ./spec/support/rubygems_ext.rb:20:in `gem_load'

Finished in 16 minutes 33 seconds (files took 0 seconds to load)
2761 examples, 1 failure, 4 pending

Failed examples:

rspec ./spec/commands/binstubs_spec.rb:232 # bundle binstubs <gem> when the gem exists in the lockfile the bundle binstub using another binstub loads all gems
```

Posting in here because it sounds like it could be the same underlying issue.

**Comment by deivid-rodriguez** [comment]
The failure in the previous comment was fixed by #3590. The failure in the initial post hasn't happened again, so I'm going to assume it's fixed. Will reopen if it happens again.
